src/day2.scala

```diff
@@ -0,0 +1,63 @@
+// using scala 3.0.2
+
+package day2
+
+import scala.io.Source
+
+@main def part1(): Unit =
+  val input = Source.fromFile("input/day2").mkString
+  val answer = part1(input)
+  println(s"The solution is $answer")
+
+@main def part2(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part2(input)
+  println(s"The solution is $answer")
+
+def part1(input: String): Int =
+  val entries = input.linesIterator.flatMap(Command.from)
+  val firstPosition = Position(0, 0)
+  val lastPosition = entries.foldLeft(firstPosition)((position, direction) =>
+    position + direction
+  )
```

Comment:
I think an interesting discussion point could be to compare this with an eta-expanded iteration function:

```suggestion
  def move(position: Position, command: Command): Position = ...
  val lastPosition = entries.foldLeft(firstPosition)(move)
```